src/encoded/renderers.py

```diff
@@ -315,3 +319,28 @@ def reload_process(process):
     args=['node', resource_filename(__name__, 'static/build/renderer.js')],
     env=node_env,
 )
+
+
+class TSVRenderer(object):
+    def __init__(self, info):
+        pass
+
+    def __call__(self, value, system):
+        """ Returns a plain TSV-encoded string with content-type
+        ``text/tsv``. The content-type may be overridden by
+        setting ``request.response.content_type``."""
+
+        request = system.get('request')
+        if request is not None:
+            response = request.response
+            ct = response.content_type
+            if ct == response.default_content_type:
+                response.content_type = 'text/tsv'
+
+        fout = io.BytesIO()
+        writer = csv.writer(fout, delimiter='\t')
+
+        writer.writerow(value.get('header', []))
+        writer.writerows(value.get('rows', []))
+
+        return fout.getvalue()
```

Comment:
I don't think it's worthwhile to make a renderer here as the functionality isn't really generic. Better just to return a Response object from the view instead. That way you can use ``response.body_file`` as the file to write to.